### PR TITLE
Service overview panel for deployment descriptor with view and copy functionality

### PR DESCRIPTION
**Pull request by dmpl1**
**Requirements**
- Add Service overview panel for deployment descriptor with a view and copy functionality

**Description of the Change**
- Added sidebar panel.
- Angular-font-awesome package is added for copy functionality. 
- Generic Copy component is created for dynamic reuse. 
- Generic Flex classes added in base.scss.
- Edit and yaml validation related functionality removed for service overview with code clean up and related css changes. 
- PR also includes filter changes of asset name in metrics and logs section.

### CODE REVIEW

File: core/jazz_ui/src/app/secondary-components/copy-element/copy-element.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+@Component({
+  selector: 'app-copy-element',
+  templateUrl: './copy-element.component.html',
+  styleUrls: ['./copy-element.component.scss']
+})
+ 
+export class CopyElementComponent implements OnInit {
+
+
+  constructor(
```
Comment (pgobin) [inline core/jazz_ui/src/app/secondary-components/copy-element/copy-element.component.ts]: Remove empty `constructor(...)` and `ngOnInit(...)` blocks.

File: core/jazz_ui/src/app/pages/environment-detail/environment-detail.component.scss [hunk 2]
```diff
@@ -158,7 +158,7 @@
   position: relative;
 }
 .page-container {
-  min-height: calc(100vh + 100px) !important;
+  // min-height: calc(100vh + 100px) !important;
```
Comment (pgobin) [inline core/jazz_ui/src/app/pages/environment-detail/environment-detail.component.scss]: Remove unused commented code.

File: core/jazz_ui/src/app/pages/environment-logs/env-logs-section.component.ts [hunk 3]
```diff
@@ -234,58 +247,76 @@ export class EnvLogsSectionComponent implements OnInit {
...
 	}
 
+	fetchAssetName(type, name) {
+		let assetName;
+		let tokens;
+		switch(type) {
+		  case 'lambda':
+		  case 'sqs':
+		  case 'iam_role':
+			tokens = name.split(':');
```
Comment (pgobin) [inline core/jazz_ui/src/app/pages/environment-logs/env-logs-section.component.ts]: Case blocks should be one indent level greater than case statement(s).

File: core/jazz_ui/src/app/pages/service-logs/service-logs.component.ts [hunk 4]
```diff
@@ -281,61 +294,78 @@ export class ServiceLogsComponent implements OnInit {
...
 	}
+	
+	fetchAssetName(type, name) {
+		let assetName;
+		let tokens;
+		switch(type) {
+		  case 'lambda':
+		  case 'sqs':
+		  case 'iam_role':
+			tokens = name.split(':');
```
Comment (pgobin) [inline core/jazz_ui/src/app/pages/service-logs/service-logs.component.ts]: Watch indentation level.

**Comment by raghits** [comment]
@dmpl1  Are the changes to Application definition in Service Overview and also changes to asset filter all in a single PR?

**Comment by devsatishm** [comment]
> @dmpl1 Are the changes to Application definition in Service Overview and also changes to asset filter all in a single PR?

PR description confirms it. @pgobin Can you validate/confirm?

**Pull request merged**